api/src/middlewares/requestValidatorMiddleware.ts

```diff
@@ -0,0 +1,31 @@
+import { UserRequest } from "../controllers/request";
+import { NextFunction, Response } from "express";
+import Joi from "joi";
+import { capture } from "../sentry";
+import { ERRORS } from "../utils";
+
+export function requestBodyValidatorMiddleware<T>(validator: Joi.ObjectSchema<T>) {
+  return (req: UserRequest, res: Response, next: NextFunction) => {
+    const payload = req.body;
+    const { error, value } = validator.validate(payload, { stripUnknown: true });
+    if (error) {
+      capture(error);
+      return res.status(400).send({ ok: false, code: ERRORS.INVALID_BODY });
+    }
+    req.validatedBody = value;
```

Comment:
```suggestion
    req.validatedPayload = value;
```
je me pose la question du naming (`validatedBody` reste cohérent biensur).

avant on avait `const { value: paylaod } = validate(...)`, du coup là on aura quelque chose comme ça
```
const { validatedBody: payload, validatedParams: { id } } = req;
```

Reply:
J'ai gardé un nommage cohérent avec RouteRequest